pints/tests/test_mcmc_adaptive.py

```diff
@@ -60,36 +58,26 @@ def setUpClass(cls):
         cls.log_posterior = pints.LogPosterior(
             cls.log_likelihood, cls.log_prior)
 
-    def test_method(self):
+    def test_instantiation(self):
 
         # Create mcmc
         x0 = self.real_parameters * 1.1
         mcmc = pints.AdaptiveCovarianceMCMC(x0)
+        self.assertEqual(0.6, mcmc.eta())
+        self.assertEqual(0.234, mcmc.target_acceptance_rate())
+        self.assertTrue(mcmc.in_initial_phase())
+
+    def test_ask_tell(self):
+
+        # ask only initialises
+        x0 = self.real_parameters * 1.1
+        mcmc = pints.AdaptiveCovarianceMCMC(x0)
+        mcmc.ask()
+        self.assertTrue(mcmc._running)
 
-        # Configure
-        mcmc.set_target_acceptance_rate(0.3)
-        mcmc.set_initial_phase(True)
-
-        # Perform short run
-        rate = []
-        chain = []
-        for i in range(100):
-            x = mcmc.ask()
-            fx = self.log_posterior(x)
-            sample = mcmc.tell(fx)
-            if i == 20:
-                mcmc.set_initial_phase(False)
-            if i >= 50:
-                chain.append(sample)
-            rate.append(mcmc.acceptance_rate())
-            if np.all(sample == x):
-                self.assertEqual(mcmc.current_log_pdf(), fx)
-
-        chain = np.array(chain)
-        rate = np.array(rate)
-        self.assertEqual(chain.shape[0], 50)
-        self.assertEqual(chain.shape[1], len(x0))
-        self.assertEqual(rate.shape[0], 100)
+        # tell
+        mcmc._proposed = None
```

Comment:
Again. The unit tester doesn't know about `._proposed`. All unit tests are of the public API. It should be possible to completely strip out all the internals and rewrite the class, without having to change the tests

Reply:
Ok, I have just copied what we were doing in the old tests. I don't see how to test this runtime error otherwise...

Reply:
Saw your comment below -- now changed. Thanks!